src/screens/PoolTogether/PoolTogetherWithdraw.js

```diff
@@ -372,92 +359,88 @@ class PoolTogetherWithdraw extends React.Component<Props, State> {
-              {(!!hasAllowance && !!withdrawPayload) &&
-                <Text center label>
-                  {t('label.feeTokenFiat', {
-                    tokenValue: withdrawPayload.feeDisplayValue,
-                    fiatValue: withdrawPayload.feeInFiat,
-                  })}
-                  {withdrawPayload.isDisabled &&
-                  // eslint-disable-next-line i18next/no-literal-string
-                  `\n${t('error.notEnoughTokenForFee', { token: withdrawPayload.feeSymbol })}`
-                  }
-                </Text>
-              }
+              {!hasAllowance && !isEstimating && !isApprovalExecuting && (
+                <Text center label>{t('poolTogetherContent.paragraph.automationMissing')}</Text>
+              )}
+              {!!isApprovalExecuting && (
+                <Text center label>{t('poolTogetherContent.paragraph.pendingAutomation')}</Text>
+              )}
+              {isEstimating && !isApprovalExecuting && !feeInfo && (
+                <Text center label>{t('label.fetchingFee')}</Text>
+              )}
             </ContentRow>
-            <ContentRow>
-              {!withdrawDisabled &&
-                <Button
-                  title={t('button.next')}
-                  onPress={() => {
-                    if (!hasAllowance && !isApprovalExecuting) {
-                      this.setState({ isAllowModalVisible: true });
-                    }
-                    return nextNavigationFunction && nextNavigationFunction();
-                  }}
-                  isLoading={isLoading}
-                  disabled={withdrawDisabled}
-                  style={{ marginBottom: 13, width: '100%' }}
-                />
-              }
-              {!!withdrawDisabled && // has to update like this so it shows the disabled style
-                <Button
-                  title={t('button.next')}
-                  disabled={withdrawDisabled}
-                  style={{ marginBottom: 13, width: '100%' }}
+            {!!withdrawTransactionAvailable && !!feeInfo && (
+              <ContentRow>
+                <FeeLabelToggle
+                  labelText={t('label.fee')}
+                  txFeeInWei={feeInfo?.fee}
+                  isLoading={isEstimating}
+                  gasToken={feeInfo?.gasToken}
+                  hasError={!!errorMessage}
+                  showFiatDefault
                 />
-              }
+              </ContentRow>
+            )}
+            {!!withdrawTransactionAvailable && !!errorMessage && (
+              <ContentRow>
+                <BaseText negative>{errorMessage}</BaseText>
+              </ContentRow>
+            )}
+            <ContentRow>
+              <Button
+                title={t('button.next')}
+                onPress={() => {
+                  if (submitDisabled) return null;
+
+                  if (!hasAllowance && !isApprovalExecuting) {
+                    this.setState({ isAllowModalVisible: true });
```

Comment:
Just wondering if it's good practice to put a setState insider a render? I understand that we have two flag evaluations so it's really unlikely it will get stuck in a re-render loop, but it's happened to me before 😂

Reply:
Just a comment, this state change looks like it's supposed to open a modal, so unless this branch will get merged into develop before the modals refactor (#2256, awaiting re-review atm), it'll have to look different anyway.

Reply:
@IAmKio it's onPress so it's not on actual render, just like any other onPress method.

Reply:
In this scenario this opens Asset allowance modal on next button press in case Asset requires allowance.

Reply:
> @IAmKio it's `onPress` so it's not on actual render, just like any other `onPress` method.

Sorry i was referring to the actual call to `setState` which would facilitate a render call - but i must have not noticed the onPress - which renders my point invalid. I'm just going to leave it there :)